lib/smart_answer/calculators/state_pension_age_calculator.rb

```diff
@@ -16,6 +16,19 @@ def state_pension_date
       StatePensionDateQuery.state_pension_date(dob, gender)
     end
 
+    def approaching_pension_age?
+      difference = (state_pension_date - Date.today).to_i
+      if difference >= 0 && difference <= claim_can_start_within
+        true
+      else
+        false
+      end
```

Comment:
This doesn't look like very idiomatic Ruby. I think it would be better to replace the whole `if` / `else` / `end` with this:

```ruby
difference >= 0 && difference <= claim_can_start_within
```